tests/dublincore.py

```diff
@@ -32,22 +36,59 @@ def testDCJSON(self):
                             % (key, UNTL_DICT))
 
     def testDCtoRDF(self):
-        '''test dublin core to RDF function'''
+        """Test Dublin Core to RDF function."""
         rdf = dcdict2rdfpy(UNTL_DICT)
         self.assertEqual(type(rdf), ConjunctiveGraph)
 
     def testDCText(self):
-        '''test that we can generate a dc text'''
+        """Test that we can generate a dc text."""
         dctext = generate_dc_txt(UNTL_DICT)
         self.assertEqual(type(dctext), str)
 
-    def testDCxml(self):
-        '''render valid dublin core xml test'''
-        dxml = generate_dc_xml(UNTL_DICT)
-        self.assertEqual(dxml.strip(), DUBLIN_CORE_XML.strip())
+    def testDCXMLRootElement(self):
+        """Check the Dublin Core for expected attrib and nsmap on root."""
+        generated_dxml = generate_dc_xml(UNTL_DICT)
+
+        # Verify root element has expected attribs, nsmap, and tag.
+        generated_dxml_root = objectify.fromstring(generated_dxml)
+
+        schema_location = ('http://www.openarchives.org/OAI/2.0/oai_dc/ '
+                           'http://www.openarchives.org/OAI/2.0/oai_dc.xsd')
+        expected_tag = '{http://www.openarchives.org/OAI/2.0/oai_dc/}dc'
+
+        self.assertTrue(schema_location in generated_dxml_root.attrib.values())
+        self.assertEqual(generated_dxml_root.nsmap, DC_NAMESPACES)
+        self.assertEqual(generated_dxml_root.tag, expected_tag)
+
+    def testDCXMLNonRootElements(self):
+        """Test for expected Dublin Core elements."""
+        generated_dxml = generate_dc_xml(UNTL_DICT)
+
+        generated_dxml_root = objectify.fromstring(generated_dxml)
+        constant_dxml_root = objectify.fromstring(DUBLIN_CORE_XML)
+        results = []
+        # Check if each of the root's children from the generated XML
+        # has a match in the string constant's XML.
+        for child in generated_dxml_root.iterchildren():
+            # For each child, get all of the children with the same tag
+            # from the XML being compared.
+            same_tag_list = constant_dxml_root.findall(child.tag)
+            element_found = False
+            child_children = len(child.getchildren())
+            # Check if any of the possible elements are a match.
+            for current in same_tag_list:
+                if (current.text == child.text and
+                        current.tail == child.tail and
+                        current.attrib == child.attrib and
+                        len(current.getchildren()) == child_children):
+                    element_found = True
+                    break
+            results.append(element_found)
```

Comment:
Good thought, though the way this test is currently now set up we only need one of the element checks to pass and then put a `True` on `results`. We expect some of the element children to fail checks because multiple ones have the same tag but different attribs and text etc., so we don't want to `extend` `results` with all comparison outcomes. I know this test is not pretty. I didn't find a particularly good way of comparing two XML blocks as the original test wanted because elements and attributes don't have guaranteed order.